todo_app/app.py

```diff
@@ -146,21 +198,46 @@ def update():
 
         return complete_item(id)
 
-    @app.route('/return_item', methods=['PUT'])
+    @ app.route('/return_item', methods=['PUT'])
     def return_item(n):
         app.logger.info('Processing default request')
         print(n)
 
         return_todo(n)
         return redirect(url_for('index'))
 
-    @app.route('/update_back', methods=['POST'])
+    @ app.route('/update_back', methods=['POST'])
     def update_back():
 
         n = request.form.get('n')
-        print(n)
+        app.logger.info("Result is %s", n)
         return return_item(n)
 
+    @ app.route('/callback')
+    def login_callback():
+        client = WebApplicationClient(webClient)
+        code = request.args.get("code")
+        client_secret = clientSecret
+        (url, headers, body) = client.prepare_token_request(
+            'https://github.com/login/oauth/access_token', code=code, client_secret=client_secret)
+        get_token_request = requests.post(url, headers=headers, data=body)
+        parse_body = client.parse_request_body_response(get_token_request.text)
+        (url, headers, body) = client.add_token("https://api.github.com/user")
+
+        request_text = requests.get(url, headers=headers, data=body).text
+
+        json_data = json.loads(request_text)
+        app.logger.info("Json data: %s", json_data)
```

Comment:
This might print out personal information which we usually don't want, except maybe at the debug level. Instead printing out the user id would be good.